### PR TITLE
[PartEditor] Update PartGraphSelector with Backend change

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise to update PartGraphSelector graph view selection state
with Backend combobox item change.

ONE-vscode-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #641, from draft #647

File: media/PartEditor/index.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ editor.Editor = class {
     // backend combobox change
     this.document.getElementById('circle-be').addEventListener('change', () => {
       this.updateDefaultCheckbox();
+      this.updateBackend();
```
Comment (seanshpark) [inline media/PartEditor/index.js:85]: (1) combobox item changed will request extension to do update backend

File: media/PartEditor/index.js [hunk 2]
```diff
@@ -135,6 +136,11 @@ editor.Editor = class {
     vscode.postMessage({command: 'updateDocument', partition: partition, opname: opname});
   }
 
+  updateBackend() {
+    let beName = this.currentBackendName();
+    vscode.postMessage({command: 'updateBackend', backend: beName});
```
Comment (seanshpark) [inline media/PartEditor/index.js:141]: (2) tell extension to update backend has changed

File: src/PartEditor/PartEditor.ts [hunk 3]
```diff
@@ -145,6 +149,10 @@ export class PartEditorProvider implements vscode.CustomTextEditorProvider, Part
 
       let content = ini.parse(this._document.getText());
       this._webview.postMessage({command: 'updatePartition', part: content});
+
+      vscode.commands.executeCommand(
```
Comment (seanshpark) [inline src/PartEditor/PartEditor.ts:153]: (3) extension tells ParGraphSelector to reflect the backend

File: src/PartEditor/PartEditor.ts [hunk 4]
```diff
@@ -270,6 +278,11 @@ export class PartEditorProvider implements vscode.CustomTextEditorProvider, Part
     }
   }
 
+  private handleUpdateBackend(backend: string) {
+    this._backEndForGraph = backend;
+    this.updateWebview();
```
Comment (seanshpark) [inline src/PartEditor/PartEditor.ts:283]: (4) refresh the document will apply selection of the backend

**Review by mhs4670go — DISMISSED** [review]
LGTM

File: src/PartEditor/PartEditor.ts [hunk 5]
```diff
@@ -114,6 +114,10 @@ export class PartEditorProvider implements vscode.CustomTextEditorProvider, Part
           console.log('selectByGraph by', message.backend);
           this.handleSelectByGraph(message.backend);
           return;
+
+        case 'updateBackend':
+          this.handleUpdateBackend(message.backend);
+          return true;
```
Comment (hyunsik-yoon) [inline src/PartEditor/PartEditor.ts:120]: other `case` just returns but this return `true`. Could you explain the reason a bit?

Comment (seanshpark) [reply src/PartEditor/PartEditor.ts:120]: typo :)
I'll remove `true`

**Review by hyunsik-yoon — APPROVED** [review]

**Comment by seanshpark** [comment]
@mhs4670go PTAL

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**